examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,297 @@
+blueprint establishes.
+
+## Architecture
+
+The following diagram shows one server and two clients that are connected to the server through a secure gRPC link:
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the NVFlare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## Deploy the blueprint
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+To deploy the NVFlare demo described in this document, you need to deploy the [Federated learning blueprint](../../../../README.md#deploy-the-blueprint) first. Then, you can deploy the NVFlare demo described in this document.
+
+### Build and push the example container image
+
+You have to create a custom image with TensorFlow and NVFlare installed. From the main `terraform` folder:
+
+1. Open [Cloud Shell](https://cloud.google.com/shell)
+
+1. Build the example container image locally on your host. It takes about 8GBs of space and can take up to 20mn to build. For a production deployment, consider using Cloud Build:
+
+   ```bash
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME="$(terraform output -raw container_image_repository_fully_qualified_hostname)"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID="${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}/$(terraform output -raw container_image_repository_name)/nvflare-tensorflow"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH="../examples/federated-learning/tff/nvflare/container-image"
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG="0.0.1"
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG=${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID}:${NVFLARE_EXAMPLE_CONTAINER_IMAGE_TAG}
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID
+
+   docker build \
+     --file "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}/Dockerfile" \
+     --tag "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}" \
+     ${NVFLARE_EXAMPLE_CONTAINER_IMAGE_BUILD_CONTEXT_PATH}
+   ```
+
+1. Authenticate Docker with the Artifact Registry repository:
+
+   ```bash
+   gcloud auth configure-docker \
+       "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}"
+   ```
+
+1. Push the container image to the Artifact Registry repository:
+
+   ```bash
+   docker image push "${NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID_WITH_TAG}"
+   ```
+
+1. Create the ConfigSync folder that will deploy the workload
+
+   ```bash
+   ACM_CONFIG_SYNC_PATH=$(terraform output -raw acm_config_sync_configuration_destination_directory_path)
```

Comment:
Are the K8s descriptors that we're going to put in this directory specific to a tenant, or cluster-wide?

If they are specific to a tenant, you should use the `acm_config_sync_tenants_configuration_destination_directory_path` and add the tenant name after that to build the path to the config sync directory that's specific to a tenant.

Reply:
It tenant' specific. I changed it to `acm_config_sync_tenants_configuration_destination_directory_path`